backend/src/v2v3/dex/data.ts

```diff
@@ -0,0 +1,182 @@
+import { bqClient } from "../../bigQuery";
+import { redisClient, getOrThrow } from "../../redisSetup";
+import { getErrorMessage, getPrice } from "./utils";
+
+const bigQueryEndpointBase = "crypto-stellar.crypto_stellar_2";
+
+async function fetchBigQueryData(query: string) {
+  try {
+    const [job] = await bqClient.createQueryJob(query);
+    console.log("running bq query:", query);
+    const [result] = await job.getQueryResults();
+    return result;
+  } catch (error: unknown) {
+    const message = getErrorMessage(error);
+    console.error("QUERY ERROR: ", message);
+    return null;
+  }
+}
+
+async function fetchMemoizedData(key: string, query: string) {
+  try {
+    const cachedData = await getOrThrow(redisClient, key);
+    const output = JSON.parse(cachedData);
+    return output;
+  } catch (error: unknown) {
+    const message = getErrorMessage(error);
+    if (message == "redis key not found") {
+      const output = await fetchBigQueryData(query);
+      await redisClient.set(key, JSON.stringify(output));
+      return output;
+    }
+    console.error("ERROR: ", message);
+  }
+}
+
+export async function getPaymentsData() {
+  const query = `
+    SELECT count(*) as data 
+    FROM ${bigQueryEndpointBase}.enriched_history_operations 
+    WHERE type in (1, 2, 13) 
+    AND closed_at >= TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL -24 HOUR) 
+    AND successful is true
+  `;
+  const payments = await fetchMemoizedData("dex-payments24h", query);
+
+  return payments[0]?.data;
+}
+
+export async function getTradeData() {
+  const tradesBase = (filter = "") => `
+    SELECT count(*) as data 
+    FROM ${bigQueryEndpointBase}.history_trades
+    ${filter};
+  `;
+  const tradesQuery24h = tradesBase(`
+    WHERE ledger_closed_at >= TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL -24 HOUR) 
```

Comment:
for these trades let's also filter out the AMM trades, adding the `AND (selling_liquidity_pool_id IS NULL OR selling_liquidity_pool_id = '') ` clause

Reply:
Should we remove the pool_id = '' from this one as well?